yt_dlp/extractor/banbye.py

```diff
@@ -0,0 +1,152 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import math
+
+from .common import InfoExtractor
+from ..compat import (
+    compat_urllib_parse_urlparse,
+    compat_parse_qs,
+)
+from ..utils import (
+    format_field,
+    unified_timestamp,
+    InAdvancePagedList,
+)
+
+
+class BanByeBaseIE(InfoExtractor):
+    _API_BASE = 'https://api.banbye.com'
+    _CDN_BASE = 'https://cdn.banbye.com'
+    _VIDEO_BASE = 'https://banbye.com/watch'
+
+    @staticmethod
+    def _extract_playlist_id(url, param='playlist'):
+        return compat_parse_qs(
+            compat_urllib_parse_urlparse(url).query).get(param, [None])[0]
+
+    def _extract_playlist(self, playlist_id):
+        data = self._download_json(f'{self._API_BASE}/playlists/{playlist_id}', playlist_id)
+        return self.playlist_result([
+            self.url_result(f'{self._VIDEO_BASE}/{video_id}', BanByeIE)
+            for video_id in data['videoIds']], playlist_id, data.get('name'))
```

Comment:
this shouldn't be fatal imo
```suggestion
            for video_id in data.get('videoIds') or []], playlist_id, data.get('name'))
```

Reply:
why not? u cant make playlist data without `videoIds`. imo an error would be better than an empty playlist when the extraction process breaks

Reply:
playlist extraction shouldn't break if videos are not avail imo. U just give a playlist with "0" entries and the core code then prints out no videos were found...that's much better

Reply:
If the key is absent from the JSON, that means the extractor is no longer correct, and this is a bug that needs to be fxed. The program should not sweep that under the rug and pretend everything is fine.

I expect an empty playlist to return an empty list in `videoIds` anyway.